theano/gpuarray/opt.py

```diff
@@ -2069,7 +2085,7 @@ def provide_inputs(self, node, inputs):
                                          node.op.subsample,
                                          node.op.filter_dilation)
 
-            result[kshape] = theano.tensor.as_tensor_variable(node.op.kshp[2:])
+            result[kshape] = theano.tensor.as_tensor_variable(node.op.kshp[-2:])
```

Comment:
I don't understand. What is not currently tested?

I don't understand what would be covered by testing the meta optimizer with the default optimizers. Do you mean the "default optimizer not in meta optimizer"? Or do you mean the default optimizer inside the meta optimizer?

If the second, then I think this is good to add if they aren't too long.